crates/matrix-sdk-crypto/src/machine.rs

```diff
@@ -129,6 +129,18 @@ pub struct OlmMachineInner {
     /// A state machine that handles creating room key backups.
     #[cfg(feature = "backups_v1")]
     backup_machine: BackupMachine,
+    /// Latest "generation" of data known by the crypto store.
+    ///
+    /// This is a counter that only increments, set in the database (and can
+    /// wrap). It's incremented whenever some process acquires a lock for the
+    /// first time. *This assumes the crypto store lock is being held, to
+    /// avoid data races on writing to this value in the store*.
+    ///
+    /// The current process will maintain this value in local memory and in the
+    /// DB over time. Observing a different value than the one read in
+    /// memory, when reading from the store indicates that somebody else has
+    /// written into the database under our feet.
+    pub(crate) crypto_store_generation: Arc<Mutex<Option<u64>>>,
```

Comment:
Does it make sense to use a `SequenceNumber` for this?

https://github.com/matrix-org/matrix-rust-sdk/blob/06600ac53fed058e3920fb9cabf840d1ea73a6d9/crates/matrix-sdk-crypto/src/store/caches.rs#L172-L182

Reply:
I could use it, and it does impl `PartialEq`/`Eq`, but I'm really just using a single `wrapping_add` here :grin: